extension/content_script.js

```diff
@@ -97,15 +98,21 @@ function startMove(dir) { // Note: I have combined the "moveUp" and "moveDown" f
 		// TODO: Consider returning boolean to see if there is any movement left
 		// If false, stop moving.
 		moveFn();
-		timer = setTimeout(repeat, speed_adj);
+		timer = setTimeout(repeat, calculateTrackerLife());
+		// Immediately fade current tracker.
+		fadeTracker();
 	})();
 }
 
+// Calculate lingering time for current tracker in ms.
+function calculateTrackerLife() {
+	return (speed * tracker.getTrackerLen()) + speed_bias;
+}
+
 // Move one sentence up
 function moveUpOne() { // Sets start and end
 	tracker.movePrevious();
 	highlight(tracker);
-	speed_adj = (speed * tracker.getTrackerLen()) + speed_bias;
```

Comment:
now that it's a getter function, less worry about recalculating this on every code path that changes speed.

Reply:
LGTM